dep/depsync/server.go

```diff
@@ -0,0 +1,67 @@
+package depsync
+
+import (
+	"github.com/go-kit/kit/endpoint"
+	"github.com/go-kit/kit/log"
+	httptransport "github.com/go-kit/kit/transport/http"
+	"github.com/gorilla/mux"
+
+	"github.com/micromdm/micromdm/pkg/httputil"
+)
+
+func NewService(syncer Syncer) *DEPSyncService {
+	return &DEPSyncService{syncer: syncer}
+}
+
+type Endpoints struct {
+	SyncNowEndpoint           endpoint.Endpoint
+	ApplyAutoAssignerEndpoint endpoint.Endpoint
+}
+
+func MakeServerEndpoints(s Service) Endpoints {
+	return Endpoints{
+		SyncNowEndpoint:           MakeSyncNowEndpoint(s),
+		ApplyAutoAssignerEndpoint: MakeApplyAutoAssignerEndpoint(s),
+	}
+}
+
+func MakeHTTPHandler(e Endpoints, logger log.Logger) *mux.Router {
+	r, options := httputil.NewRouter(logger)
+
+	// POST		/v1/dep/syncnow			request a DEP sync operation to happen now
+	// POST		/v1/dep/autoassigner	set a DEP auto-assigner
+
+	// POST		/v1/dep/profiles		get a DEP profile given a known profile UUID
+	// GET		/v1/dep/account			get information about the dep account
+	// POST		/v1/dep/devices			get device details given a list of serials
+
+	r.Methods("POST").Path("/v1/dep/syncnow").Handler(httptransport.NewServer(
+		e.SyncNowEndpoint,
+		decodeEmptyRequest,
+		encodeEmptyResponse,
+		options...,
+	))
+
+	r.Methods("POST").Path("/v1/dep/autoassigners").Handler(httptransport.NewServer(
+		e.ApplyAutoAssignerEndpoint,
+		decodeApplyAutoAssignerRequest,
+		httputil.EncodeJSONResponse,
+		options...,
+	))
+
+	// r.Methods("GET").Path("/v1/dep/account").Handler(httptransport.NewServer(
```

Comment:
why are these commented out?

Reply:
No reason, they shouldn't be there. They're vestiges from the service (dep) I copied them from.